digraph "CFG forkotlinx.coroutines.InterruptibleKt.runInterruptibleInExpectedContext(Lkotlin\/coroutines\/CoroutineContext;Lkotlin\/jvm\/functions\/Function0;)Ljava\/lang\/Object;" {
Node_14 [shape=record,label="{14\:\ ?|MTH_ENTER_BLOCK\l}"];
Node_26 [shape=record,label="{26\:\ 0x0001|SYNTHETIC\lEXC_TOP_SPLITTER\l}"];
Node_16 [shape=record,label="{16\:\ 0x0001|TRY_ENTER\lTRY_LEAVE\lTryCatch #1 \{InterruptedException \-\> 0x001b, blocks: (B:16:0x0001, B:18:0x0012, B:17:0x000e), top: B:26:0x0001 \}\lCatch: InterruptedException \-\> 0x001b\l|0x0001: NEW_INSTANCE  (r0 I:kotlinx.coroutines.ThreadState) =  kotlinx.coroutines.ThreadState A[Catch: InterruptedException \-\> 0x001b, TRY_ENTER] (LINE:48)\l0x0003: INVOKE  (r1 I:kotlinx.coroutines.Job) = (r3 I:kotlin.coroutines.CoroutineContext A[D('coroutineContext' kotlin.coroutines.CoroutineContext)]) type: STATIC call: kotlinx.coroutines.JobKt.getJob(kotlin.coroutines.CoroutineContext):kotlinx.coroutines.Job A[Catch: InterruptedException \-\> 0x001b, MD:(kotlin.coroutines.CoroutineContext):kotlinx.coroutines.Job (m)]\l0x0007: INVOKE  (r0 I:kotlinx.coroutines.ThreadState), (r1 I:kotlinx.coroutines.Job) type: DIRECT call: kotlinx.coroutines.ThreadState.\<init\>(kotlinx.coroutines.Job):void A[Catch: InterruptedException \-\> 0x001b, MD:(kotlinx.coroutines.Job):void (m)]\l0x000a: INVOKE  (r0 I:kotlinx.coroutines.ThreadState A[D('threadState' kotlinx.coroutines.ThreadState)]) type: VIRTUAL call: kotlinx.coroutines.ThreadState.setup():void A[Catch: InterruptedException \-\> 0x001b, MD:():void (m), TRY_LEAVE] (LINE:49)\l}"];
Node_17 [shape=record,label="{17\:\ 0x000e|TRY_ENTER\lTRY_LEAVE\lTryCatch #1 \{InterruptedException \-\> 0x001b, blocks: (B:16:0x0001, B:18:0x0012, B:17:0x000e), top: B:26:0x0001 \}\lCatch: all \-\> 0x0016\l|0x000e: INVOKE  (r1 I:java.lang.Object) = (r4 I:kotlin.jvm.functions.Function0 A[D('block' kotlin.jvm.functions.Function0)]) type: INTERFACE call: kotlin.jvm.functions.Function0.invoke():java.lang.Object A[Catch: all \-\> 0x0016, MD:():R (m), TRY_ENTER, TRY_LEAVE] (LINE:51)\l}"];
Node_18 [shape=record,label="{18\:\ 0x0012|TRY_ENTER\lTryCatch #1 \{InterruptedException \-\> 0x001b, blocks: (B:16:0x0001, B:18:0x0012, B:17:0x000e), top: B:26:0x0001 \}\lCatch: InterruptedException \-\> 0x001b\l|0x0012: INVOKE  (r0 I:kotlinx.coroutines.ThreadState A[D('threadState' kotlinx.coroutines.ThreadState)]) type: VIRTUAL call: kotlinx.coroutines.ThreadState.clearInterrupt():void A[Catch: InterruptedException \-\> 0x001b, MD:():void (m), TRY_ENTER] (LINE:53)\l}"];
Node_19 [shape=record,label="{19\:\ 0x0015|RETURN\l|0x0015: RETURN  (r1 I:T)  (LINE:51)\l}"];
Node_15 [shape=record,label="{15\:\ ?|MTH_EXIT_BLOCK\l}"];
Node_27 [shape=record,label="{27\:\ ?|SYNTHETIC\lEXC_BOTTOM_SPLITTER\l}"];
Node_23 [shape=record,label="{23\:\ 0x001b|ExcHandler: InterruptedException \-\> 0x001b\l|0x001b: MOVE_EXCEPTION  (r0 I:java.lang.InterruptedException A[CUSTOM_DECLARE]) =  A[DONT_INLINE, ExcHandler: InterruptedException \-\> 0x001b] (LINE:55)\l}"];
Node_24 [shape=record,label="{24\:\ 0x001c|0x001c: NEW_INSTANCE  (r1 I:java.util.concurrent.CancellationException) =  java.util.concurrent.CancellationException  (LINE:56)\l0x001e: CONST_STR  (r2 I:java.lang.String) =  \"Blocking call was interrupted due to parent cancellation\" \l0x0020: INVOKE  (r1 I:java.util.concurrent.CancellationException), (r2 I:java.lang.String) type: DIRECT call: java.util.concurrent.CancellationException.\<init\>(java.lang.String):void A[MD:(java.lang.String):void (c)]\l0x0023: MOVE  (r2 I:??[OBJECT, ARRAY]) = (r0 I:??[OBJECT, ARRAY] A[D('e' java.lang.InterruptedException)]) \l0x0024: CHECK_CAST (r2 I:java.lang.Throwable) = (java.lang.Throwable) (r2 I:??[OBJECT, ARRAY]) \l0x0026: INVOKE  (r1 I:java.lang.Throwable) = (r1 I:java.util.concurrent.CancellationException), (r2 I:java.lang.Throwable) type: VIRTUAL call: java.util.concurrent.CancellationException.initCause(java.lang.Throwable):java.lang.Throwable A[MD:(java.lang.Throwable):java.lang.Throwable (s)]\l}"];
Node_25 [shape=record,label="{25\:\ 0x002a|0x002a: THROW  (r1 I:java.lang.Throwable) \l}"];
MethodNode[shape=record,label="{public static final T kotlinx.coroutines.InterruptibleKt.runInterruptibleInExpectedContext((r3 I:kotlin.coroutines.CoroutineContext A[D('coroutineContext' kotlin.coroutines.CoroutineContext), IMMUTABLE_TYPE, METHOD_ARGUMENT]), (r4 I:kotlin.jvm.functions.Function0\<? extends T\> A[D('block' kotlin.jvm.functions.Function0), IMMUTABLE_TYPE, METHOD_ARGUMENT]))  | USE_LINES_HINTS\lCROSS_EDGE: B:25:0x002a \-\> B:15:?, CROSS_EDGE: B:26:0x0001 \-\> B:23:0x001b\lINLINE_NOT_NEEDED\lTryCatch #1 \{InterruptedException \-\> 0x001b, blocks: (B:16:0x0001, B:18:0x0012, B:17:0x000e), top: B:26:0x0001 \}\lTYPE_VARS: [T]\lSIGNATURE: \<T:Ljava\/lang\/Object;\>(Lkotlin\/coroutines\/CoroutineContext;Lkotlin\/jvm\/functions\/Function0\<+TT;\>;)TT;\lDebug Info:\l  \-1 \-0x001a: r3 'coroutineContext' Lkotlin\/coroutines\/CoroutineContext;\l  \-1 \-0x001a: r4 'block' Lkotlin\/jvm\/functions\/Function0;\l  0x000a\-0x001b: r0 'threadState' Lkotlinx\/coroutines\/ThreadState;\l  0x001c\-0x002a: r0 'e' Ljava\/lang\/InterruptedException;\l  0x001b\-0x002a: r3 'coroutineContext' Lkotlin\/coroutines\/CoroutineContext;\l  0x001b\-0x002a: r4 'block' Lkotlin\/jvm\/functions\/Function0;\l}"];
MethodNode -> Node_14;Node_14 -> Node_26;
Node_26 -> Node_16;
Node_26 -> Node_23;
Node_16 -> Node_17;
Node_17 -> Node_18;
Node_18 -> Node_19;
Node_18 -> Node_27;
Node_19 -> Node_15;
Node_27 -> Node_23;
Node_23 -> Node_24;
Node_24 -> Node_25;
Node_25 -> Node_15;
}

